lang/java/reef-utils/src/test/java/org/apache/reef/util/MultiAsyncToSyncTest.java

```diff
@@ -0,0 +1,240 @@
+    try {
+      LOG.log(Level.INFO, "Call getting task result...");
+      result = task.get();
+    } catch (Exception e) {
+      result = -1;
+    }
+    return result;
+  }
+
+  /**
+   * Insure all test tasks have completed.
+   */
+  public void complete() {
+    try {
+      for (FutureTask<Integer> task : taskQueue) {
+        task.get();
+      }
+    } catch (Exception e) {
+      LOG.log(Level.INFO, "Caught exception waiting for completion...");
+    }
+  }
+}
+
+/**
+ * Verify proper operation of the MultiAsyncToSync class.
+ */
+public final class MultiAsyncToSyncTest {
+  private static final Logger LOG = Logger.getLogger(MultiAsyncToSyncTest.class.getName());
+
+  /**
+   * Verify calculations successfully complete when no timeout occurs.
+   */
+  @Test
+  public void testNoTimeout() {
+    LOG.log(Level.INFO, "Starting...");
+
+    // Parameters that do not force a timeout.
+    final int incrementerSleepTimeSeconds = 2;
+    final long timeoutPeriodSeconds = 4;
+    final Integer input = 1;
+
+    SynchronousApi apiObject = new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);
+    final Integer result = apiObject.apiCall(input);
+    apiObject.complete();
+
+    Assert.assertTrue(result.equals(input + 1));
+  }
+
+  /**
+   * Verify an error is returned when a timeout occurs.
+   */
+  @Test
+  public void testTimeout() {
+    LOG.log(Level.INFO, "Starting...");
+
+    // Parameters that do not force a timeout.
+    final int incrementerSleepTimeSeconds = 4;
+    final long timeoutPeriodSeconds = 2;
+    final Integer input = 1;
+
+    SynchronousApi apiObject = new SynchronousApi(incrementerSleepTimeSeconds, timeoutPeriodSeconds);
+    final Integer result = apiObject.apiCall(input);
+    apiObject.complete();
+
+    Assert.assertTrue(result.equals(0));
+  }
+
+  /**
+   * Used to make multiple asynchronous calls to the synchronous API.
+   */
+  final class AsynchronousCaller implements Callable<Boolean> {
+    private final SynchronousApi apiObject;
+    private final Integer input;
+
+    /**
+     * @param apiObject The synchronous api to be calle.
+     * @param input Input value for the api call.
+     */
+    AsynchronousCaller(final SynchronousApi apiObject, final Integer input) {
+      this.apiObject = apiObject;
+      this.input = input;
+    }
+
+    /**
+     * @return A Boolean object that indicates whether
+     *         or not the processing was successful.
+     * @throws Exception
+     */
+    public Boolean call() throws Exception {
+      final Integer result = apiObject.apiCall(input);
+      return result.equals(input + 1);
+    }
+  }
+
+  /**
+   * Verify no interaction occurs when multiple calls are in flight.
+   */
+  @Test
+  public void testMulitpleCalls() {
+    LOG.log(Level.INFO, "Starting...");
+
+    // Parameters that do not force a timeout.
+    final int incrementerSleepTimeSeconds = 2;
+    final long timeoutPeriodSeconds = 4;
+    final Integer input = 1;
```

Comment:
Since boxed was required in one place I used it everywhere for that variable. Do prefer I limit the boxing to only where it is absolutely needed?